### PR TITLE
Add the ability to customize the HTTP user agent

### PR DESCRIPTION
**Pull request by steiza**
To help with things like https://github.com/sigstore/sigstore-go/issues/143.

See https://github.com/sigstore/sigstore-go/pull/216 for an example of how this would be used.

### CODE REVIEW

**Review by mnm678 — APPROVED** [review]

**Review by haydentherapper — APPROVED** [review]

**Comment by steiza** [comment]
Just FYI, even thought this pull request has an approval, I don't have the ability to merge it as I don't have write access to the repository. I think we need a go-tuf maintainer to merge this pull request.

**Comment by MDr164** [comment]
Done

**Pull request merged**
